From aa983beb86fb3307566458fcbcd7e571a031ce7a Mon Sep 17 00:00:00 2001
From: Andrew Gunnerson <andrewgunnerson@gmail.com>
Date: Fri, 2 Jan 2015 14:27:31 -0500
Subject: [PATCH] Fixes

- Build with the ndk toolchain
- Build all libraries and binaries statically
- Build binaries using gcc instead of g++ to avoid linking libstdc++.so
---
 e2fsck/Android.mk     |  44 +++----------
 lib/blkid/Android.mk  |  40 +-----------
 lib/e2p/Android.mk    |  29 +--------
 lib/et/Android.mk     |  31 +---------
 lib/ext2fs/Android.mk |  44 +------------
 lib/quota/Android.mk  |  27 +-------
 lib/uuid/Android.mk   |  41 +------------
 misc/Android.mk       | 167 ++++++++++----------------------------------------
 resize/Android.mk     |  23 ++-----
 9 files changed, 56 insertions(+), 390 deletions(-)

diff --git a/e2fsck/Android.mk b/e2fsck/Android.mk
index b4b2aef..b5b982d 100644
--- a/e2fsck/Android.mk
+++ b/e2fsck/Android.mk
@@ -7,12 +7,10 @@ libext2_profile_src_files :=  \
 	prof_err.c \
 	profile.c
 
-libext2_profile_shared_libraries := \
+libext2_profile_static_libraries := \
 	libext2_com_err
 
-libext2_profile_system_shared_libraries := libc
-
-libext2_profile_c_includes := external/e2fsprogs/lib
+libext2_profile_c_includes := lib
 
 libext2_profile_cflags := -O2 -g -W -Wall \
 	-DHAVE_UNISTD_H \
@@ -44,25 +42,13 @@ libext2_profile_cflags := -O2 -g -W -Wall \
 include $(CLEAR_VARS)
 
 LOCAL_SRC_FILES := $(libext2_profile_src_files)
-LOCAL_SYSTEM_SHARED_LIBRARIES := $(libext2_profile_system_shared_libraries)
-LOCAL_SHARED_LIBRARIES := $(libext2_profile_shared_libraries)
+LOCAL_STATIC_LIBRARIES := $(libext2_profile_static_libraries)
 LOCAL_C_INCLUDES := $(libext2_profile_c_includes)
 LOCAL_CFLAGS := $(libext2_profile_cflags)
 LOCAL_MODULE := libext2_profile
 LOCAL_MODULE_TAGS := optional
 
-include $(BUILD_SHARED_LIBRARY)
-
-include $(CLEAR_VARS)
-
-LOCAL_SRC_FILES := $(libext2_profile_src_files)
-LOCAL_SHARED_LIBRARIES := $(addsuffix _host, $(libext2_profile_shared_libraries))
-LOCAL_C_INCLUDES := $(libext2_profile_c_includes)
-LOCAL_CFLAGS := $(libext2_profile_cflags)
-LOCAL_MODULE := libext2_profile_host
-LOCAL_MODULE_TAGS := optional
-
-include $(BUILD_HOST_SHARED_LIBRARY)
+include $(BUILD_STATIC_LIBRARY)
 
 #########################
 # Build the e2fsck binary
@@ -96,7 +82,7 @@ e2fsck_src_files :=  \
 	region.c \
 	sigcatcher.c
 
-e2fsck_shared_libraries := \
+e2fsck_static_libraries := \
 	libext2fs \
 	libext2_blkid \
 	libext2_uuid \
@@ -104,9 +90,8 @@ e2fsck_shared_libraries := \
 	libext2_quota \
 	libext2_com_err \
 	libext2_e2p
-e2fsck_system_shared_libraries := libc
 
-e2fsck_c_includes := external/e2fsprogs/lib
+e2fsck_c_includes := lib
 
 e2fsck_cflags := -O2 -g -W -Wall -fno-strict-aliasing \
 	-DHAVE_DIRENT_H \
@@ -145,20 +130,9 @@ include $(CLEAR_VARS)
 LOCAL_SRC_FILES := $(e2fsck_src_files)
 LOCAL_C_INCLUDES := $(e2fsck_c_includes)
 LOCAL_CFLAGS := $(e2fsck_cflags)
-LOCAL_SYSTEM_SHARED_LIBRARIES := $(e2fsck_system_shared_libraries)
-LOCAL_SHARED_LIBRARIES := $(e2fsck_shared_libraries)
+LOCAL_STATIC_LIBRARIES := $(e2fsck_static_libraries)
 LOCAL_MODULE := e2fsck
 LOCAL_MODULE_TAGS := optional
+TARGET_CXX := $(TOOLCHAIN_PREFIX)gcc
+LOCAL_LDFLAGS := -static
 include $(BUILD_EXECUTABLE)
-
-include $(CLEAR_VARS)
-
-LOCAL_SRC_FILES := $(e2fsck_src_files)
-LOCAL_C_INCLUDES := $(e2fsck_c_includes)
-LOCAL_CFLAGS := $(e2fsck_cflags)
-LOCAL_SHARED_LIBRARIES := $(addsuffix _host, $(e2fsck_shared_libraries))
-LOCAL_MODULE := e2fsck_host
-LOCAL_MODULE_STEM := e2fsck
-LOCAL_MODULE_TAGS := optional
-
-include $(BUILD_HOST_EXECUTABLE)
diff --git a/lib/blkid/Android.mk b/lib/blkid/Android.mk
index 33be570..5607743 100644
--- a/lib/blkid/Android.mk
+++ b/lib/blkid/Android.mk
@@ -15,15 +15,9 @@ libext2_blkid_src_files := \
 	version.c \
 
 
-libext2_blkid_shared_libraries := libext2_uuid
+libext2_blkid_static_libraries := libext2_uuid
 
-libext2_blkid_system_shared_libraries := libc
-
-libext2_blkid_static_libraries := libext2_uuid_static
-
-libext2_blkid_system_static_libraries := libc
-
-libext2_blkid_c_includes := external/e2fsprogs/lib
+libext2_blkid_c_includes := lib
 
 libext2_blkid_cflags := -O2 -g -W -Wall -fno-strict-aliasing \
 	-DHAVE_UNISTD_H \
@@ -56,38 +50,10 @@ libext2_blkid_cflags_linux := \
 include $(CLEAR_VARS)
 
 LOCAL_SRC_FILES := $(libext2_blkid_src_files)
-LOCAL_SYSTEM_SHARED_LIBRARIES := $(libext2_blkid_system_shared_libraries)
-LOCAL_SHARED_LIBRARIES := $(libext2_blkid_shared_libraries)
+LOCAL_STATIC_LIBRARIES := $(libext2_blkid_static_libraries)
 LOCAL_C_INCLUDES := $(libext2_blkid_c_includes)
 LOCAL_CFLAGS := $(libext2_blkid_cflags) $(libext2_blkid_cflags_linux) -fno-strict-aliasing
 LOCAL_MODULE := libext2_blkid
 LOCAL_MODULE_TAGS := optional
 
-include $(BUILD_SHARED_LIBRARY)
-
-include $(CLEAR_VARS)
-
-LOCAL_SRC_FILES := $(libext2_blkid_src_files)
-LOCAL_STATIC_LIBRARIES := $(libext2_blkid_static_libraries) $(libext2_blkid_system_static_libraries)
-LOCAL_C_INCLUDES := $(libext2_blkid_c_includes)
-LOCAL_CFLAGS := $(libext2_blkid_cflags) $(libext2_blkid_cflags_linux) -fno-strict-aliasing
-LOCAL_PRELINK_MODULE := false
-LOCAL_MODULE := libext2_blkid
-LOCAL_MODULE_TAGS := optional
-
 include $(BUILD_STATIC_LIBRARY)
-
-include $(CLEAR_VARS)
-
-LOCAL_SRC_FILES := $(libext2_blkid_src_files)
-LOCAL_SHARED_LIBRARIES := $(addsuffix _host, $(libext2_blkid_shared_libraries))
-LOCAL_C_INCLUDES := $(libext2_blkid_c_includes)
-ifeq ($(HOST_OS),linux)
-LOCAL_CFLAGS := $(libext2_blkid_cflags) $(libext2_blkid_cflags_linux)
-else
-LOCAL_CFLAGS := $(libext2_blkid_cflags)
-endif
-LOCAL_MODULE := libext2_blkid_host
-LOCAL_MODULE_TAGS := optional
-
-include $(BUILD_HOST_SHARED_LIBRARY)
diff --git a/lib/e2p/Android.mk b/lib/e2p/Android.mk
index 829774d..5c44915 100644
--- a/lib/e2p/Android.mk
+++ b/lib/e2p/Android.mk
@@ -22,7 +22,7 @@ libext2_e2p_src_files := \
 	ostype.c \
 	percent.c
 
-libext2_e2p_c_includes := external/e2fsprogs/lib
+libext2_e2p_c_includes := lib
 
 libext2_e2p_cflags := -O2 -g -W -Wall \
 	-DHAVE_UNISTD_H \
@@ -50,39 +50,12 @@ libext2_e2p_cflags := -O2 -g -W -Wall \
         -DHAVE_SYS_PARAM_H \
 	-DHAVE_SYSCONF
 
-libext2_e2p_system_shared_libraries := libc
-
-libext2_e2p_system_static_libraries := libc
-
 include $(CLEAR_VARS)
 
 LOCAL_SRC_FILES := $(libext2_e2p_src_files)
 LOCAL_C_INCLUDES := $(libext2_e2p_c_includes)
 LOCAL_CFLAGS := $(libext2_e2p_cflags)
-LOCAL_SYSTEM_SHARED_LIBRARIES := $(libext2_e2p_system_shared_libraries)
-LOCAL_MODULE := libext2_e2p
-LOCAL_MODULE_TAGS := optional
-
-include $(BUILD_SHARED_LIBRARY)
-
-include $(CLEAR_VARS)
-
-LOCAL_SRC_FILES := $(libext2_e2p_src_files)
-LOCAL_C_INCLUDES := $(libext2_e2p_c_includes)
-LOCAL_CFLAGS := $(libext2_e2p_cflags)
-LOCAL_STATIC_LIBRARIES := $(libext2_e2p_system_static_libraries)
-LOCAL_PRELINK_MODULE := false
 LOCAL_MODULE := libext2_e2p
 LOCAL_MODULE_TAGS := optional
 
 include $(BUILD_STATIC_LIBRARY)
-
-include $(CLEAR_VARS)
-
-LOCAL_SRC_FILES := $(libext2_e2p_src_files)
-LOCAL_C_INCLUDES := $(libext2_e2p_c_includes)
-LOCAL_CFLAGS := $(libext2_e2p_cflags)
-LOCAL_MODULE := libext2_e2p_host
-LOCAL_MODULE_TAGS := optional
-
-include $(BUILD_HOST_SHARED_LIBRARY)
diff --git a/lib/et/Android.mk b/lib/et/Android.mk
index 256f310..ee9f8dd 100644
--- a/lib/et/Android.mk
+++ b/lib/et/Android.mk
@@ -7,7 +7,7 @@ libext2_com_err_src_files := \
 	com_err.c \
 	com_right.c
 
-libext2_com_err_c_includes := external/e2fsprogs/lib
+libext2_com_err_c_includes := lib
 
 libext2_com_err_cflags := -O2 -g -W -Wall \
 	-DHAVE_UNISTD_H \
@@ -37,41 +37,12 @@ libext2_com_err_cflags_linux := \
 	-DHAVE_LSEEK64 \
 	-DHAVE_LSEEK64_PROTOTYPE
 
-libext2_com_err_system_shared_libraries := libc
-
-include $(CLEAR_VARS)
-
-LOCAL_SRC_FILES := $(libext2_com_err_src_files)
-LOCAL_C_INCLUDES := $(libext2_com_err_c_includes)
-LOCAL_CFLAGS := $(libext2_com_err_cflags) $(libext2_com_err_cflags_linux)
-LOCAL_SYSTEM_SHARED_LIBRARIES := libc
-LOCAL_MODULE := libext2_com_err
-LOCAL_MODULE_TAGS := optional
-
-include $(BUILD_SHARED_LIBRARY)
-
 include $(CLEAR_VARS)
 
 LOCAL_SRC_FILES := $(libext2_com_err_src_files)
 LOCAL_C_INCLUDES := $(libext2_com_err_c_includes)
 LOCAL_CFLAGS := $(libext2_com_err_cflags) $(libext2_com_err_cflags_linux)
-LOCAL_STATIC_LIBRARIES := libc
 LOCAL_MODULE := libext2_com_err
 LOCAL_MODULE_TAGS := optional
-LOCAL_PRELINK_MODULE := false
 
 include $(BUILD_STATIC_LIBRARY)
-
-include $(CLEAR_VARS)
-
-LOCAL_SRC_FILES := $(libext2_com_err_src_files)
-LOCAL_C_INCLUDES := $(libext2_com_err_c_includes)
-ifeq ($(HOST_OS),linux)
-LOCAL_CFLAGS := $(libext2_com_err_cflags) $(libext2_com_err_cflags_linux)
-else
-LOCAL_CFLAGS := $(libext2_com_err_cflags)
-endif
-LOCAL_MODULE := libext2_com_err_host
-LOCAL_MODULE_TAGS := optional
-
-include $(BUILD_HOST_SHARED_LIBRARY)
diff --git a/lib/ext2fs/Android.mk b/lib/ext2fs/Android.mk
index 11c558f..0ea0e9f 100644
--- a/lib/ext2fs/Android.mk
+++ b/lib/ext2fs/Android.mk
@@ -72,23 +72,13 @@ libext2fs_src_files := \
 # get rid of this?!
 libext2fs_src_files += test_io.c
 
-libext2fs_shared_libraries := \
-	libext2_com_err \
-	libext2_uuid \
-	libext2_blkid \
-	libext2_e2p
-
-libext2fs_system_shared_libraries := libc
-
 libext2fs_static_libraries := \
 	libext2_com_err \
-	libext2_uuid_static \
+	libext2_uuid \
 	libext2_blkid \
 	libext2_e2p
 
-libext2fs_system_static_libraries := libc
-
-libext2fs_c_includes := external/e2fsprogs/lib
+libext2fs_c_includes := lib
 
 libext2fs_cflags := -O2 -g -W -Wall \
 	-DHAVE_UNISTD_H \
@@ -121,38 +111,10 @@ libext2fs_cflags_linux := \
 include $(CLEAR_VARS)
 
 LOCAL_SRC_FILES := $(libext2fs_src_files)
-LOCAL_SYSTEM_SHARED_LIBRARIES := $(libext2fs_system_shared_libraries)
-LOCAL_SHARED_LIBRARIES := $(libext2fs_shared_libraries)
+LOCAL_STATIC_LIBRARIES := $(libext2fs_static_libraries)
 LOCAL_C_INCLUDES := $(libext2fs_c_includes)
 LOCAL_CFLAGS := $(libext2fs_cflags) $(libext2fs_cflags_linux)
 LOCAL_MODULE := libext2fs
 LOCAL_MODULE_TAGS := optional
 
-include $(BUILD_SHARED_LIBRARY)
-
-include $(CLEAR_VARS)
-
-LOCAL_SRC_FILES := $(libext2fs_src_files)
-LOCAL_STATIC_LIBRARIES := $(libext2fs_static_libraries) $(libext2fs_system_static_libraries)
-LOCAL_C_INCLUDES := $(libext2fs_c_includes)
-LOCAL_CFLAGS := $(libext2fs_cflags) $(libext2fs_cflags_linux)
-LOCAL_PRELINK_MODULE := false
-LOCAL_MODULE := libext2fs
-LOCAL_MODULE_TAGS := optional
-
 include $(BUILD_STATIC_LIBRARY)
-
-include $(CLEAR_VARS)
-
-LOCAL_SRC_FILES := $(libext2fs_src_files)
-LOCAL_SHARED_LIBRARIES := $(addsuffix _host, $(libext2fs_shared_libraries))
-LOCAL_C_INCLUDES := $(libext2fs_c_includes)
-ifeq ($(HOST_OS),linux)
-LOCAL_CFLAGS := $(libext2fs_cflags) $(libext2fs_cflags_linux)
-else
-LOCAL_CFLAGS := $(libext2fs_cflags)
-endif
-LOCAL_MODULE := libext2fs_host
-LOCAL_MODULE_TAGS := optional
-
-include $(BUILD_HOST_SHARED_LIBRARY)
diff --git a/lib/quota/Android.mk b/lib/quota/Android.mk
index 74a6a07..006b284 100644
--- a/lib/quota/Android.mk
+++ b/lib/quota/Android.mk
@@ -7,7 +7,7 @@ libext2_quota_src_files := \
 	quotaio_v2.c \
 	../../e2fsck/dict.c
 
-libext2_quota_c_includes := external/e2fsprogs/lib
+libext2_quota_c_includes := lib
 
 libext2_quota_cflags := -O2 -g -W -Wall \
 	-DHAVE_UNISTD_H \
@@ -35,8 +35,6 @@ libext2_quota_cflags := -O2 -g -W -Wall \
         -DHAVE_SYS_PARAM_H \
 	-DHAVE_SYSCONF
 
-libext2_quota_shared_libraries := libext2fs libext2_com_err
-
 libext2_quota_static_libraries := libext2fs libext2_com_err
 
 include $(CLEAR_VARS)
@@ -44,31 +42,8 @@ include $(CLEAR_VARS)
 LOCAL_SRC_FILES := $(libext2_quota_src_files)
 LOCAL_C_INCLUDES := $(libext2_quota_c_includes)
 LOCAL_CFLAGS := $(libext2_quota_cflags)
-LOCAL_SYSTEM_SHARED_LIBRARIES := libc $(libext2_quota_shared_libraries)
-LOCAL_MODULE := libext2_quota
-LOCAL_MODULE_TAGS := optional
-
-include $(BUILD_SHARED_LIBRARY)
-
-include $(CLEAR_VARS)
-
-LOCAL_SRC_FILES := $(libext2_quota_src_files)
-LOCAL_C_INCLUDES := $(libext2_quota_c_includes)
-LOCAL_CFLAGS := $(libext2_quota_cflags)
 LOCAL_STATIC_LIBRARIES := libc $(libext2_quota_static_libraries)
-LOCAL_PRELINK_MODULE := false
 LOCAL_MODULE := libext2_quota
 LOCAL_MODULE_TAGS := optional
 
 include $(BUILD_STATIC_LIBRARY)
-
-include $(CLEAR_VARS)
-
-LOCAL_SRC_FILES := $(libext2_quota_src_files)
-LOCAL_C_INCLUDES := $(libext2_quota_c_includes)
-LOCAL_CFLAGS := $(libext2_quota_cflags)
-LOCAL_MODULE := libext2_quota_host
-LOCAL_MODULE_TAGS := optional
-LOCAL_SHARED_LIBRARIES := $(addsuffix _host, $(libext2_quota_shared_libraries))
-
-include $(BUILD_HOST_SHARED_LIBRARY)
diff --git a/lib/uuid/Android.mk b/lib/uuid/Android.mk
index a43a361..10a8bb9 100644
--- a/lib/uuid/Android.mk
+++ b/lib/uuid/Android.mk
@@ -13,7 +13,7 @@ libext2_uuid_src_files := \
 	uuid_time.c
 
 
-libext2_uuid_c_includes := external/e2fsprogs/lib
+libext2_uuid_c_includes := lib
 
 libext2_uuid_cflags := -O2 -g -W -Wall \
 	-DHAVE_INTTYPES_H \
@@ -42,52 +42,13 @@ libext2_uuid_cflags := -O2 -g -W -Wall \
         -DHAVE_SYS_PARAM_H \
 	-DHAVE_SYSCONF
 
-libext2_uuid_system_shared_libraries := libc
-
 include $(CLEAR_VARS)
 
 LOCAL_SRC_FILES := $(libext2_uuid_src_files)
 LOCAL_C_INCLUDES := $(libext2_uuid_c_includes)
 LOCAL_CFLAGS := $(libext2_uuid_cflags)
 LOCAL_EXPORT_C_INCLUDE_DIRS := $(LOCAL_PATH)
-LOCAL_SYSTEM_SHARED_LIBRARIES := $(libext2_uuid_system_shared_libraries)
 LOCAL_MODULE := libext2_uuid
 LOCAL_MODULE_TAGS := optional
 
-include $(BUILD_SHARED_LIBRARY)
-
-include $(CLEAR_VARS)
-
-LOCAL_SRC_FILES := $(libext2_uuid_src_files)
-LOCAL_C_INCLUDES := $(libext2_uuid_c_includes)
-LOCAL_CFLAGS := $(libext2_uuid_cflags)
-LOCAL_EXPORT_C_INCLUDE_DIRS := $(LOCAL_PATH)
-LOCAL_MODULE := libext2_uuid_host
-LOCAL_MODULE_TAGS := optional
-
-include $(BUILD_HOST_SHARED_LIBRARY)
-
-
-include $(CLEAR_VARS)
-
-LOCAL_SRC_FILES := $(libext2_uuid_src_files)
-LOCAL_C_INCLUDES := $(libext2_uuid_c_includes)
-LOCAL_CFLAGS := $(libext2_uuid_cflags)
-LOCAL_EXPORT_C_INCLUDE_DIRS := $(LOCAL_PATH)
-LOCAL_STATIC_LIBRARIES := libc
-LOCAL_MODULE := libext2_uuid_static
-LOCAL_MODULE_TAGS := optional
-
 include $(BUILD_STATIC_LIBRARY)
-
-
-include $(CLEAR_VARS)
-
-LOCAL_SRC_FILES := $(libext2_uuid_src_files)
-LOCAL_C_INCLUDES := $(libext2_uuid_c_includes)
-LOCAL_CFLAGS := $(libext2_uuid_cflags)
-LOCAL_EXPORT_C_INCLUDE_DIRS := $(LOCAL_PATH)
-LOCAL_MODULE := libext2_uuid_host
-LOCAL_MODULE_TAGS := optional
-
-include $(BUILD_HOST_STATIC_LIBRARY)
diff --git a/misc/Android.mk b/misc/Android.mk
index 645fc9c..f924788 100644
--- a/misc/Android.mk
+++ b/misc/Android.mk
@@ -8,8 +8,8 @@ mke2fs_src_files := \
 	default_profile.c
 
 mke2fs_c_includes := \
-	external/e2fsprogs/lib \
-	external/e2fsprogs/e2fsck
+	lib \
+	e2fsck
 
 mke2fs_cflags := -O2 -g -W -Wall \
 	-DHAVE_UNISTD_H \
@@ -43,7 +43,7 @@ mke2fs_cflags_linux := \
 
 mke2fs_cflags += -DNO_CHECK_BB
 
-mke2fs_shared_libraries := \
+mke2fs_static_libraries := \
 	libext2fs \
 	libext2_blkid \
 	libext2_uuid \
@@ -52,35 +52,18 @@ mke2fs_shared_libraries := \
 	libext2_com_err \
 	libext2_e2p
 
-mke2fs_system_shared_libraries := libc
-
 include $(CLEAR_VARS)
 
 LOCAL_SRC_FILES := $(mke2fs_src_files)
 LOCAL_C_INCLUDES := $(mke2fs_c_includes)
 LOCAL_CFLAGS := $(mke2fs_cflags) $(mke2fs_cflags_linux)
-LOCAL_SYSTEM_SHARED_LIBRARIES := $(mke2fs_system_shared_libraries)
-LOCAL_SHARED_LIBRARIES := $(mke2fs_shared_libraries)
+LOCAL_STATIC_LIBRARIES := $(mke2fs_static_libraries)
 LOCAL_MODULE := mke2fs
 LOCAL_MODULE_TAGS := optional
+TARGET_CXX := $(TOOLCHAIN_PREFIX)gcc
+LOCAL_LDFLAGS := -static
 include $(BUILD_EXECUTABLE)
 
-include $(CLEAR_VARS)
-
-LOCAL_SRC_FILES := $(mke2fs_src_files)
-LOCAL_C_INCLUDES := $(mke2fs_c_includes)
-ifeq ($(HOST_OS),linux)
-LOCAL_CFLAGS := $(mke2fs_cflags) $(mke2fs_cflags_linux)
-else
-LOCAL_CFLAGS := $(mke2fs_cflags)
-endif
-LOCAL_SHARED_LIBRARIES := $(addsuffix _host, $(mke2fs_shared_libraries))
-LOCAL_MODULE := mke2fs_host
-LOCAL_MODULE_STEM := mke2fs
-LOCAL_MODULE_TAGS := optional
-
-include $(BUILD_HOST_EXECUTABLE)
-
 ###########################################################################
 # Build tune2fs
 #
@@ -89,8 +72,8 @@ tune2fs_src_files := \
 	util.c
 
 tune2fs_c_includes := \
-	external/e2fsprogs/lib \
-	external/e2fsprogs/e2fsck
+	lib \
+	e2fsck
 
 tune2fs_cflags := -O2 -g -W -Wall \
 	-DHAVE_UNISTD_H \
@@ -122,74 +105,28 @@ tune2fs_cflags := -O2 -g -W -Wall \
 
 tune2fs_cflags += -DNO_CHECK_BB
 
-tune2fs_shared_libraries := \
+tune2fs_static_libraries := \
 	libext2fs \
 	libext2_com_err \
 	libext2_blkid \
 	libext2_quota \
 	libext2_uuid \
-	libext2_e2p
-
-tune2fs_system_shared_libraries := libc
-
-
-tune2fs_static_libraries := \
-	libext2_com_err \
-	libext2_blkid \
-	libext2_quota \
-	libext2_uuid_static \
 	libext2_e2p \
 	libext2fs
 
-tune2fs_system_static_libraries := libc
-
 include $(CLEAR_VARS)
 
 LOCAL_SRC_FILES := $(tune2fs_src_files)
 LOCAL_C_INCLUDES := $(tune2fs_c_includes)
 LOCAL_CFLAGS := $(tune2fs_cflags)
-LOCAL_SHARED_LIBRARIES := $(tune2fs_shared_libraries)
-LOCAL_SYSTEM_SHARED_LIBRARIES := $(tune2fs_system_shared_libraries)
+LOCAL_STATIC_LIBRARIES := $(tune2fs_static_libraries)
 LOCAL_MODULE := tune2fs
 LOCAL_MODULE_TAGS := optional
+TARGET_CXX := $(TOOLCHAIN_PREFIX)gcc
+LOCAL_LDFLAGS := -static
 
 include $(BUILD_EXECUTABLE)
 
-include $(CLEAR_VARS)
-
-LOCAL_SRC_FILES := $(tune2fs_src_files)
-LOCAL_C_INCLUDES := $(tune2fs_c_includes)
-LOCAL_CFLAGS := $(tune2fs_cflags)
-LOCAL_STATIC_LIBRARIES := $(tune2fs_static_libraries) $(tune2fs_system_static_libraries)
-LOCAL_FORCE_STATIC_EXECUTABLE := true
-LOCAL_MODULE := tune2fs_static
-LOCAL_MODULE_TAGS := optional
-
-include $(BUILD_EXECUTABLE)
-
-include $(CLEAR_VARS)
-
-LOCAL_SRC_FILES := $(tune2fs_src_files)
-LOCAL_C_INCLUDES := $(tune2fs_c_includes)
-LOCAL_CFLAGS := $(tune2fs_cflags) -DBUILD_AS_LIB
-LOCAL_STATIC_LIBRARIES := $(tune2fs_static_libraries) $(tune2fs_system_static_libraries)
-LOCAL_MODULE := libtune2fs
-LOCAL_MODULE_TAGS := optional
-
-include $(BUILD_STATIC_LIBRARY)
-
-include $(CLEAR_VARS)
-
-LOCAL_SRC_FILES := $(tune2fs_src_files)
-LOCAL_C_INCLUDES := $(tune2fs_c_includes)
-LOCAL_CFLAGS := $(tune2fs_cflags)
-LOCAL_SHARED_LIBRARIES := $(addsuffix _host, $(tune2fs_shared_libraries))
-LOCAL_MODULE := tune2fs_host
-LOCAL_MODULE_STEM := tune2fs
-LOCAL_MODULE_TAGS := optional
-
-include $(BUILD_HOST_EXECUTABLE)
-
 #########################################################################
 # Build badblocks
 #
@@ -199,7 +136,7 @@ badblocks_src_files := \
 	badblocks.c
 
 badblocks_c_includes := \
-	external/e2fsprogs/lib
+	lib
 
 badblocks_cflags := -O2 -g -W -Wall \
 	-DHAVE_UNISTD_H \
@@ -229,39 +166,26 @@ badblocks_cflags := -O2 -g -W -Wall \
         -DHAVE_SYS_PARAM_H \
 	-DHAVE_SYSCONF
 
-badblocks_shared_libraries := \
+badblocks_static_libraries := \
 	libext2fs \
 	libext2_com_err \
 	libext2_uuid \
 	libext2_blkid \
 	libext2_e2p
 
-badblocks_system_shared_libraries := libc
-
 include $(CLEAR_VARS)
 
 LOCAL_SRC_FILES := $(badblocks_src_files)
 LOCAL_C_INCLUDES := $(badblocks_c_includes)
 LOCAL_CFLAGS := $(badblocks_cflags)
-LOCAL_SHARED_LIBRARIES := $(badblocks_shared_libraries)
-LOCAL_SYSTEM_SHARED_LIBRARIES := $(badblocks_system_shared_libraries)
+LOCAL_STATIC_LIBRARIES := $(badblocks_static_libraries)
 LOCAL_MODULE := badblocks
 LOCAL_MODULE_TAGS := optional
+TARGET_CXX := $(TOOLCHAIN_PREFIX)gcc
+LOCAL_LDFLAGS := -static
 
 include $(BUILD_EXECUTABLE)
 
-include $(CLEAR_VARS)
-
-LOCAL_SRC_FILES := $(badblocks_src_files)
-LOCAL_C_INCLUDES := $(badblocks_c_includes)
-LOCAL_CFLAGS := $(badblocks_cflags)
-LOCAL_SHARED_LIBRARIES := $(addsuffix _host, $(badblocks_shared_libraries))
-LOCAL_MODULE := badblocks_host
-LOCAL_MODULE_STEM := badblocks
-LOCAL_MODULE_TAGS := optional
-
-include $(BUILD_HOST_EXECUTABLE)
-
 #########################################################################
 # Build chattr
 #
@@ -271,7 +195,7 @@ chattr_src_files := \
 	chattr.c
 
 chattr_c_includes := \
-	external/e2fsprogs/lib
+	lib
 
 chattr_cflags := -O2 -g -W -Wall \
 	-DHAVE_UNISTD_H \
@@ -301,36 +225,23 @@ chattr_cflags := -O2 -g -W -Wall \
         -DHAVE_SYS_PARAM_H \
 	-DHAVE_SYSCONF
 
-chattr_shared_libraries := \
+chattr_static_libraries := \
 	libext2_com_err \
 	libext2_e2p
 
-chattr_system_shared_libraries := libc
-
 include $(CLEAR_VARS)
 
 LOCAL_SRC_FILES := $(chattr_src_files)
 LOCAL_C_INCLUDES := $(chattr_c_includes)
 LOCAL_CFLAGS := $(chattr_cflags)
-LOCAL_SHARED_LIBRARIES := $(chattr_shared_libraries)
-LOCAL_SYSTEM_SHARED_LIBRARIES := $(chattr_system_shared_libraries)
+LOCAL_STATIC_LIBRARIES := $(chattr_static_libraries)
 LOCAL_MODULE := chattr
 LOCAL_MODULE_TAGS := optional
+TARGET_CXX := $(TOOLCHAIN_PREFIX)gcc
+LOCAL_LDFLAGS := -static
 
 include $(BUILD_EXECUTABLE)
 
-include $(CLEAR_VARS)
-
-LOCAL_SRC_FILES := $(chattr_src_files)
-LOCAL_C_INCLUDES := $(chattr_c_includes)
-LOCAL_CFLAGS := $(chattr_cflags)
-LOCAL_SHARED_LIBRARIES := $(addsuffix _host, $(chattr_shared_libraries))
-LOCAL_MODULE := chattr_host
-LOCAL_MODULE_STEM := chattr
-LOCAL_MODULE_TAGS := optional
-
-include $(BUILD_HOST_EXECUTABLE)
-
 #########################################################################
 # Build lsattr
 #
@@ -340,7 +251,7 @@ lsattr_src_files := \
 	lsattr.c
 
 lsattr_c_includes := \
-	external/e2fsprogs/lib
+	lib
 
 lsattr_cflags := -O2 -g -W -Wall \
 	-DHAVE_UNISTD_H \
@@ -370,36 +281,23 @@ lsattr_cflags := -O2 -g -W -Wall \
         -DHAVE_SYS_PARAM_H \
 	-DHAVE_SYSCONF
 
-lsattr_shared_libraries := \
+lsattr_static_libraries := \
 	libext2_com_err \
 	libext2_e2p
 
-lsattr_system_shared_libraries := libc
-
 include $(CLEAR_VARS)
 
 LOCAL_SRC_FILES := $(lsattr_src_files)
 LOCAL_C_INCLUDES := $(lsattr_c_includes)
 LOCAL_CFLAGS := $(lsattr_cflags)
-LOCAL_SHARED_LIBRARIES := $(lsattr_shared_libraries)
-LOCAL_SYSTEM_SHARED_LIBRARIES := $(lsattr_system_shared_libraries)
+LOCAL_STATIC_LIBRARIES := $(lsattr_static_libraries)
 LOCAL_MODULE := lsattr
 LOCAL_MODULE_TAGS := optional
+TARGET_CXX := $(TOOLCHAIN_PREFIX)gcc
+LOCAL_LDFLAGS := -static
 
 include $(BUILD_EXECUTABLE)
 
-include $(CLEAR_VARS)
-
-LOCAL_SRC_FILES := $(lsattr_src_files)
-LOCAL_C_INCLUDES := $(lsattr_c_includes)
-LOCAL_CFLAGS := $(lsattr_cflags)
-LOCAL_SHARED_LIBRARIES := $(addsuffix _host, $(lsattr_shared_libraries))
-LOCAL_MODULE := lsattr_host
-LOCAL_MODULE_STEM := lsattr
-LOCAL_MODULE_TAGS := optional
-
-include $(BUILD_HOST_EXECUTABLE)
-
 #########################################################################
 # Build blkid
 #
@@ -409,7 +307,7 @@ blkid_src_files := \
     blkid.c
 
 blkid_c_includes := \
-    external/e2fsprogs/lib
+    lib
 
 blkid_cflags := -O2 -g -W -Wall \
     -DHAVE_UNISTD_H \
@@ -440,22 +338,21 @@ blkid_cflags := -O2 -g -W -Wall \
     -DHAVE_SYSCONF \
     -DHAVE_TERMIO_H
 
-blkid_shared_libraries := \
+blkid_static_libraries := \
     libext2fs \
     libext2_blkid \
     libext2_com_err \
     libext2_e2p
 
-blkid_system_shared_libraries := libc
-
 include $(CLEAR_VARS)
 
 LOCAL_SRC_FILES := $(blkid_src_files)
 LOCAL_C_INCLUDES := $(blkid_c_includes)
 LOCAL_CFLAGS := $(blkid_cflags)
-LOCAL_SHARED_LIBRARIES := $(blkid_shared_libraries)
-LOCAL_SYSTEM_SHARED_LIBRARIES := $(blkid_system_shared_libraries)
+LOCAL_STATIC_LIBRARIES := $(blkid_static_libraries)
 LOCAL_MODULE := blkid
 LOCAL_MODULE_TAGS := optional
+TARGET_CXX := $(TOOLCHAIN_PREFIX)gcc
+LOCAL_LDFLAGS := -static
 
 include $(BUILD_EXECUTABLE)
diff --git a/resize/Android.mk b/resize/Android.mk
index 41f9fe0..8f3a492 100644
--- a/resize/Android.mk
+++ b/resize/Android.mk
@@ -8,7 +8,7 @@ resize2fs_src_files := \
 	sim_progress.c \
 	resource_track.c
 
-resize2fs_c_includes := external/e2fsprogs/lib
+resize2fs_c_includes := lib
 
 resize2fs_cflags := -O2 -g -W -Wall \
 	-DHAVE_UNISTD_H \
@@ -36,34 +36,21 @@ resize2fs_cflags := -O2 -g -W -Wall \
         -DHAVE_SYS_PARAM_H \
 	-DHAVE_SYSCONF
 
-resize2fs_shared_libraries := \
+resize2fs_static_libraries := \
 	libext2fs \
 	libext2_com_err \
 	libext2_e2p \
 	libext2_uuid \
 	libext2_blkid
 
-resize2fs_system_shared_libraries := libc
-
 include $(CLEAR_VARS)
 
 LOCAL_SRC_FILES := $(resize2fs_src_files)
 LOCAL_C_INCLUDES := $(resize2fs_c_includes)
 LOCAL_CFLAGS := $(resize2fs_cflags)
-LOCAL_SHARED_LIBRARIES := $(resize2fs_shared_libraries)
-LOCAL_SYSTEM_SHARED_LIBRARIES := $(resize2fs_system_shared_libraries)
+LOCAL_STATIC_LIBRARIES := $(resize2fs_static_libraries)
 LOCAL_MODULE := resize2fs
 LOCAL_MODULE_TAGS := optional
+TARGET_CXX := $(TOOLCHAIN_PREFIX)gcc
+LOCAL_LDFLAGS := -static
 include $(BUILD_EXECUTABLE)
-
-include $(CLEAR_VARS)
-
-LOCAL_SRC_FILES := $(resize2fs_src_files)
-LOCAL_C_INCLUDES := $(resize2fs_c_includes)
-LOCAL_CFLAGS := $(resize2fs_cflags)
-LOCAL_SHARED_LIBRARIES := $(addsuffix _host, $(resize2fs_shared_libraries))
-LOCAL_MODULE := resize2fs_host
-LOCAL_MODULE_STEM := resize2fs
-LOCAL_MODULE_TAGS := optional
-
-include $(BUILD_HOST_EXECUTABLE)
-- 
2.5.0

